code/core/src/manager/components/sidebar/Sidebar.tsx

```diff
@@ -99,6 +120,21 @@ const useCombination = (
   return useMemo(() => ({ hash, entries: Object.entries(hash) }), [hash]);
 };
 
+const updateQueryParams = (params: Record<string, string>) => {
+  const url = new URL(window.location.href);
+  Object.entries(params).forEach(([key, value]) => {
+    if (value) {
+      url.searchParams.set(key, value);
+    } else {
+      url.searchParams.delete(key);
+    }
+  });
+  window.history.pushState({}, '', url);
+};
```

Comment:
FYI I've needed to do something like that in the past in a preview context (for [this token doc](https://styleguide.backmarket.com/system/main/index.html?path=%2Fdocs%2Ffoundations-colors-raw--docs&q=grey-100&s=color-raw-grey-100) though it appears to have recently broken).

It could be useful to isolate this into an exported util alongside the navigate function, having both together would help advanced users build more interactive docs.

Reply:
Turns out it's already in the `api` object. Updating the code.